src/main/java/net/mcreator/ui/component/JFileSelector.java

```diff
@@ -0,0 +1,141 @@
+ */
+
+package net.mcreator.ui.component;
+
+import javafx.stage.FileChooser;
+import net.mcreator.ui.component.util.PanelUtils;
+import net.mcreator.ui.dialogs.file.FileChooserType;
+import net.mcreator.ui.dialogs.file.FileDialogs;
+import net.mcreator.ui.dialogs.preferences.PreferencesDialog;
+import net.mcreator.ui.init.L10N;
+import net.mcreator.ui.init.UIRES;
+import net.mcreator.util.StringUtils;
+
+import javax.swing.*;
+import java.awt.*;
+import java.io.File;
+import java.util.EventObject;
+import java.util.function.Consumer;
+
+public class JFileSelector extends JPanel {
+
+	private File value;
+	private final boolean isFolder;
+	private final Window parent;
+	private final boolean allowNullValue;
+	private final Consumer<EventObject> fct;
+	private final FileChooser.ExtensionFilter[] filters;
+
+	/**
+	 * Create a custom file selector component to allow or not the usage of a null value for the desired folder.
+	 *
+	 * @param parent         The file chooser's parent
+	 * @param defaultValue   The value that will be set to the component at its creation
+	 * @param allowNullValue If true, another button will be added to make the returned value null
+	 * @param fct            This is mainly used to change the status of the {@link PreferencesDialog} buttons, so we can click on "Apply"
+	 */
+	public JFileSelector(Window parent, File defaultValue, boolean allowNullValue, Consumer<EventObject> fct) {
+		this.isFolder = true;
+		this.value = defaultValue;
+		this.parent = parent;
+		this.allowNullValue = allowNullValue;
+		this.fct = fct;
+		this.filters = new FileChooser.ExtensionFilter[] {};
+
+		createComponent();
+	}
+
+	/**
+	 * Create a custom file selector component to allow or not the usage of a null value for the desired file.
+	 *
+	 * @param parent         The file chooser's parent
+	 * @param defaultValue   The value that will be set to the component at its creation
+	 * @param allowNullValue If true, another button will be added to make the returned value null
+	 * @param fct            This is mainly used to change the status of the {@link PreferencesDialog} buttons, so we can click on "Apply"
+	 * @param filters        If the user needs to choose a file, you can define the file chooser's filters
+	 */
+	public JFileSelector(Window parent, File defaultValue, boolean allowNullValue, Consumer<EventObject> fct,
+			FileChooser.ExtensionFilter... filters) {
+		this.isFolder = false;
+		this.value = defaultValue;
+		this.parent = parent;
+		this.allowNullValue = allowNullValue;
+		this.fct = fct;
+		this.filters = filters;
+
+		createComponent();
+	}
+
+	private void createComponent() {
+		setLayout(new BorderLayout(2, 0));
+		setOpaque(false);
+
+		JTextField pathField = new JTextField(15);
+		pathField.setEditable(false);
+		pathField.setHorizontalAlignment(JTextField.CENTER);
+		pathField.setOpaque(true);
+
+		if (value == null || value.getAbsolutePath().endsWith("Not specified"))

```

Comment:
You should not store "Not specified" value but just use "null" in those cases, coming back to the only open problem in the PR

Reply:
The serialization of nill is due to the Gson builder, but there are other problems aftervto fix or to make sure that don't happen.

Reply:
null is not serialized, but that is not a problem, as if field is not present in json, it will be properly deserialized back to null

Reply:
```suggestion
		if (value == null)
```